File: leetcode/3427-sum-of-variable-length-subarrays/3427-sum-of-variable-length-subarrays.py
class Solution:
    def subarraySum(self, nums: List[int]) -> int:
        n = len(nums)
        prefix = [0] * (n + 1)
        total = 0

        for i in range(n):
            prefix[i + 1] = prefix[i] + nums[i]

        for i in range(n):
            start = max(0 , i - nums[i])
            total += prefix[i + 1] - prefix[start]

        return total

File: LeetCode/Easy/3427-sum-of-variable-length-subarrays/3427-sum-of-variable-length-subarrays.java
class Solution {
    public int subarraySum(int[] nums) {
        int len = nums.length;
        int result = 0;
        int[] prefix = new int[len + 1];

        for (int i = 0; i < len; i++) {
            prefix[i + 1] = prefix[i] + nums[i];
            
            int start = Math.max(0, i - nums[i]);
            result += prefix[i + 1] - prefix[start];
        }

        return result;
    }
}